native/src/routes/settings/components/Settings.js

```diff
@@ -78,6 +83,10 @@ export default class Settings extends React.Component<PropsType, StateType> {
     }
   }
 
+  componentWillUnmount() {
+    this.unsubscribeNavigationListener()
+  }
```

Comment:
Why is this needed? why do we need to load settings when the comonent unmounts?

Reply:
In settings screen we see if tracking is enabled or disabled, when clicking on tracking-settings and navigate to jpal-settings-screen, we can update if tracking is enabled and when nativate.goBack to general settings, there is still the old value, because settings are only loaded in constructor. Therefore i add an listener for "on-focus" so component loads settings when opend (also on goBack()). We have to unsubscribe from this listener when unmounting.